utils/cloudinfo/aws.go

```diff
@@ -23,7 +23,10 @@ import (
 )
 
 func onAWS() bool {
-	client := ec2metadata.New(session.New(&aws.Config{}))
+	// by default, we set retries to 0 for this operation.
+	// when not running on AWS, not setting retries can cause the operation to block
```

Comment:
I am going to clarify the comment.

The default aws behavior is to retry 4 times, each with a 5 second timeout.  If you run cadvisor outside of gce or aws, you end up waiting 20s because the cloud detection checks gce first, aws second, and then azure third, and then finally it says I give up.